### PR TITLE
triage ItMiiAuxiliaryImage test failure in release/3.3

### PR DESCRIPTION
**Pull request by xiancao**
Triage ItMiiAuxiliaryImage test failure in release/3.3:

Jenkins result:
https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/8981/
https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/8978/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java [hunk 1]
```diff
@@ -353,15 +352,35 @@ public static void checkDomainEventContainsExpectedMsg(String opNamespace,
...
+
+  private static Callable<Boolean> domainEventContainsExpectedMsg(String opNamespace,
+                                                                  String domainNamespace,
+                                                                  String domainUid,
+                                                                  String reason,
+                                                                  String type,
+                                                                  OffsetDateTime timestamp,
+                                                                  String expectedMsg) {
+    return () -> {
+      if (domainEventExists(opNamespace, domainNamespace, domainUid, reason, type, timestamp)) {
```
Comment (doxiao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java:375]: The checkDomainEventContainsExpectedMsg() method only checks the first event that matches the event type, reason, timestamp and etc. Multiple DOMAIN_PROCESSING_FAILED type events could be generated during the test time frame. The logic need to make sure that it checks all matching events.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java [hunk 2]
```diff
@@ -217,6 +218,37 @@ public static CoreV1Event getEvent(String opNamespace, String domainNamespace, S
...
+   */
+  public static List<CoreV1Event> getEvents(String domainNamespace,
+                                            String reason,
+                                            String type,
+                                            OffsetDateTime timestamp) {
+
+    List<CoreV1Event> events = new ArrayList<>();
+
+    try {
+      List<CoreV1Event> allEvents = Kubernetes.listNamespacedEvents(domainNamespace);
```
Comment (doxiao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java:237]: Use listOpGeneratedNamespacedEvents to only get operator generated events.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java (hunk 1 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java (hunk 2 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java:237]: where can I can this listOpGeneratedNamespacedEvents method?

Comment (doxiao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java:239]: Change listNamespacedEvents to listOpGeneratedNamespacedEvents. Both are utility methods in Kubernetes.java.

Comment (doxiao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java:239]: Sorry, I just realized that this is for release/3.3, where the new method does not exist.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java [hunk 3]
```diff
@@ -218,6 +218,42 @@ public static CoreV1Event getEvent(String opNamespace, String domainNamespace, S
...
+      for (CoreV1Event event : allEvents) {
+        Map<String, String> labels = event.getMetadata().getLabels();
+        if (event.getReason().equals(reason)
+            && (isEqualOrAfter(timestamp, event))
+            && event.getType().equals(type)
+            && labels.containsKey("weblogic.createdByOperator")
+            && labels.get("weblogic.domainUID").equals(domainUid)) {
```
Comment (doxiao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java:246]: The list contains all of the events in the namespace, we don't know if event event has labels. So we need to make sure that the "labels" variable is not null. Not sure if it is easier to add the listOpGeneratedNamespacedEvents method into release/3.3.

**Review by doxiao — APPROVED** [review]
LGTM.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java (hunk 2 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java]: added listOpGeneratedNamespacedEvents in Kubernetes.java

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java [hunk 4]
```diff
@@ -217,6 +218,37 @@ public static CoreV1Event getEvent(String opNamespace, String domainNamespace, S
...
+                                            String reason,
+                                            String type,
+                                            OffsetDateTime timestamp) {
+
+    List<CoreV1Event> events = new ArrayList<>();
+
+    try {
+      List<CoreV1Event> allEvents = Kubernetes.listNamespacedEvents(domainNamespace);
+      for (CoreV1Event event : allEvents) {
+        if (event.getReason().equals(reason) && (isEqualOrAfter(timestamp, event))) {
```
[the remark this answers is not in the document]
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java (hunk 3 above)
Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java:247]: fixed

**Review by vanajamukkara — APPROVED** [review]
LG

**Pull request merged**